web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.NOT_STARTED)).toBeInTheDocument();
+      });
+    });
+
+    it("updates the task progress when the modal is saved", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByTestId("modal-button-primary"));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.COMPLETED)).toBeInTheDocument();
+      });
+    });
+
+    it("user gets redirected to roadmap with the proper query parameter", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        profileData: generateProfileData({ businessPersona: "STARTING" }),
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByTestId("modal-button-primary"));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.COMPLETED)).toBeInTheDocument();
+      });
+      expect(mockPush).toHaveBeenCalledWith({
+        pathname: ROUTES.roadmap,
+        query: { fromFormBusinessEntity: "false", fromTaxRegistration: "true" },
+      });
+    });
+
+    it("shows the business name field if it's not populated for non sp/gp", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "",
+          legalStructureId: "limited-liability-company",
+        }),
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
```

Comment:
the test name says we're looking at the business name field... why are we asserting about existing employees?